src/objc/ModalWebViewController.mm

```diff
@@ -30,7 +30,24 @@ - (void)viewDidLoad {
   WKProcessPool *processPool = [[WKProcessPool alloc] init];
   configuration.processPool = processPool;
 
-  // Create a WKWebsiteDataStore
+  // Add user content controller for window.close override
+  WKUserContentController *userContentController = [[WKUserContentController alloc] init];
+
+  // Inject JavaScript to override window.close
+    NSString *js = @"window.close = function() { window.webkit.messageHandlers.windowCloseCalled.postMessage('noop'); };";
+
+  WKUserScript *script = [[WKUserScript alloc] initWithSource:js
+                                                injectionTime:WKUserScriptInjectionTimeAtDocumentStart
+                                             forMainFrameOnly:NO];
+    [userContentController addUserScript:script];
+    [userContentController addScriptMessageHandler:self name:@"windowCloseCalled"];
+    configuration.userContentController = userContentController;
+    
+  // Add native handler for window.close call
+  [userContentController addScriptMessageHandler:self name:@"closeHandler"];
```

Comment:
Is `closeHandler` a standard JS API to intercept `window.close`? How does this work?